keyboards/royal_kludge/rk61/keymaps/default/keymap.h

```diff
@@ -0,0 +1,50 @@
+        {   KC_TAB,     KC_Q,       KC_W,       KC_E,       KC_R,       KC_T,       KC_Y,       KC_U,       KC_I,       KC_O,       KC_P,       KC_LBRC,    KC_RBRC,    KC_BSLS,    }, \
+        {   caps,       KC_A,       KC_S,       KC_D,       KC_F,       KC_G,       KC_H,       KC_J,       KC_K,       KC_L,       KC_SCLN,    KC_QUOT,    nuhs,       KC_ENT,     }, \
+        {   KC_LSFT,    nubs,       KC_Z,       KC_X,       KC_C,       KC_V,       KC_B,       KC_N,       KC_M,       KC_COMM,    KC_DOT,     KC_UP,      KC_RSFT,    XXXXXXX,    }, \
+        {   KC_LCTL,    KC_LGUI,    KC_LALT,    XXXXXXX,    XXXXXXX,    XXXXXXX,    KC_SPC,     XXXXXXX,    XXXXXXX,    KC_LEFT,    KC_DOWN,    KC_RIGHT,   XXXXXXX,    MO(_FN),    }  \
+    }
+#define BASE_KEYMAP BASE_KEYMAP_(KC_CAPS, KC_NUHS, KC_NUBS)
+
+// snap right hand modifiers: Alt, context menu, Ctrl instead of arrows
+#define RMODS_KEYMAP_(ralt) { \
+        {   _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    }, \
+        {   _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    }, \
+        {   _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    }, \
+        {   _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    KC_SLSH,    _______,    _______,    }, \
+        {   _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    ralt,       KC_APP,     KC_RCTL,    _______,    _______,    }  \
+    }
+
+#define RMODS_KEYMAP RMODS_KEYMAP_(KC_RALT)
+
+// functional layer (F1-F10, backlight controls etc.)
+#define FN_KEYMAP { \
+        {   KC_GRV,     KC_F1,      KC_F2,      KC_F3,      KC_F4,      KC_F5,      KC_F6,      KC_F7,      KC_F8,      KC_F9,      KC_F10,     KC_F11,     KC_F12,     KC_DEL,     }, \
+        {   _______,    KC_F13,     KC_F14,     KC_F15,     KC_F16,     KC_F17,     KC_PSCR,    KC_SLCK,    KC_PAUS,    _______,    RGB_HUI,    RGB_SAI,    RGB_VAI,    RGB_MOD,    }, \
+        {   RESET,      KC_VOLU,    KC_BRIU,    _______,    _______,    _______,    KC_INS,     KC_HOME,    KC_PGUP,    _______,    RGB_SPD,    RGB_SPI,    RGB_MOD,    TG(_RMODS), }, \
+        {   _______,    _______,    KC_VOLD,    KC_BRID,    _______,    _______,    _______,    KC_DEL,     KC_END,     KC_PGDN,    _______,    KC_SLSH,    _______,    _______,    }, \
+        {   _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    KC_RALT,    KC_APP,     KC_RCTL,    _______,    _______,    }  \
```

Comment:
Try adding GUI lock toggle. If you can, also try implementing that the key will illuminate to white when GUI lock is turned on.

```suggestion
        {   _______,    GUI_TOG,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    KC_RALT,    KC_APP,     KC_RCTL,    _______,    _______,    }  \
```

Reply:
I don't know how to handle the `GUI_TOG`. How can I get the status of the GUI keys? You can see how `caps_lock` will be handled:
```c
void rgb_matrix_indicators_advanced_user(uint8_t led_min, uint8_t led_max) {
    led_t led_state = host_keyboard_led_state();

    if (led_state.caps_lock) {
        // Set capslock key to orange (capslock is led number defined as `CL`)
        rgb_matrix_set_color(CL, 255, 130, 15);
    }
}
```
But the `led_t` struct does not contain the info about GUI. :-(

Have you an idea how to get this state inside of `rgb_matrix_indicators_advanced_user`?

Reply:
I've found a solution. :-)
```c
    if (keymap_config.no_gui) {
        // Set GUI key to lemon (GUI is led number defined as `GL`)
        rgb_matrix_set_color(GL, 255, 230, 15);
    }
```